lib/services.credentials.js

```diff
@@ -42,44 +45,82 @@ bedrock.events.on('bedrock-express.configure.routes', function(app) {
       },
       */
       get: function(req, res, callback) {
-        var identityId = req.user.identity.id;
-        var idHash = database.hash(identityId);
-        var query = {};
-        // default to non-deleted offers
-        query['credential.sysStatus'] = {$ne: 'deleted'};
-        // FIXME: default query recipient to current id?
-        if(req.query.recipient) {
-          query.recipient = database.hash(req.query.recipient);
-        }
-        if(req.query.issuer) {
-          query.issuer = database.hash(req.query.issuer);
-        }
-        if(req.query.filter) {
-          query['credential.sysState'] = req.query.filter;
-        }
-        if(req.query.id) {
-          // FIXME: use _getCredentialRecord()
-          store.get(
-            req.user.identity, req.query.id,
-            function(err, credential, meta) {
-              if(err) {
-                return callback(err);
-              }
-              _credentialToFormat(credential, meta, req.query.format, callback);
-            });
+        // request is not authenticated
+        if(!req.user.identity && req.query.id) {
+          // FIXME: return a credential marked as public
+          // if no public crendentials, respond with 404
+          var publicQuery = {
+            'credential.id': req.query.id,
+            'credential.sysPublic': '*'
+          };
+          return store.getAll(null, publicQuery, function(err, records) {
+            if(records.length === 0) {
+              return callback(new BedrockError(
+                'Credential not found.', 'NotFound', {
+                  public: true,
+                  httpStatusCode: 404
+                }));
+            }
+            var credential = records[0].credential;
+            delete credential.sysPublic;
+            // add in flag to know this is public credential data
+            // FIXME: better way to convey this meta info?
+            credential.sysIsPublic = true;
+            return callback(null, credential);
+          });
+
         } else {
```

Comment:
Unnecessary else; looks like there is a short-cutting return earlier. Remove else and unindent.